File: codes/ruby/chapter06/answer_A45.rb
# 入力
N, C = gets.chomp.split
A = gets.chomp
N = N.to_i

# スコアの計算
score = 0
N.times do |i|
  case A[i]
  when 'W'
    score += 0
  when 'B'
    score += 1
  when 'R'
    score += 2
  end
end

# 出力
if score % 3 == 0 && C == 'W'
  puts "Yes"
elsif score % 3 == 1 && C == 'B'
  puts "Yes"
elsif score % 3 == 2 && C == 'R'
  puts "Yes"
else
  puts "No"
end

File: codes/go/chapter06/answer_A45.go
package main

import (
	"fmt"
)

func main() {
	var N int
	var C string
	var A string
	fmt.Scan(&N, &C, &A)

	score := 0
	for i := 0; i < N; i++ {
		if A[i] == 'W' {
			score += 0
		} else if A[i] == 'B' {
			score += 1
		} else if A[i] == 'R' {
			score += 2
		}
	}

	if score%3 == 0 && C == "W" {
		fmt.Println("Yes")
	} else if score%3 == 1 && C == "B" {
		fmt.Println("Yes")
	} else if score%3 == 2 && C == "R" {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}
